astropy/table/tests/test_table.py

```diff
@@ -1372,7 +1372,8 @@ def test_equality_masked():
     assert np.all(t == t)
 
     # Assert no rows are different
-    assert not np.any(t != t)
+    t_filled = t.filled()
+    assert not np.any(t_filled != t_filled)
```

Comment:
I don't think this is the right fix - instead we should fix ``Table.__eq__`` (see also my comment in https://github.com/astropy/astropy/pull/7946#issuecomment-432230204)

Reply:
OK, I'll look into `__eq__`. Thanks for all the useful comments!